client/my-sites/stats/jetpack-upsell-section/index.tsx

```diff
@@ -23,15 +25,40 @@ const QUERY_VALUES = {
 	checkoutBackUrl: window.location.href,
 };
 
+function shouldHideUpsellSection( purchases: Purchase[] ) {
+	if ( purchases.length === 0 ) {
+		return false;
+	}
+
+	// Filter out expired plans.
+	const plans = purchases.filter( ( purchase ) => purchase.expiryStatus !== 'expired' );
```

Comment:
Is importing the `filterPurchasesByProducts` or `areProductsOwned` here a better way than creating another similar check?

Reply:
I've updated the code to use `areProductsOwned` with the existing `const` strings.